app/models/post.rb

```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+  after_save :update_counter
```

Comment:
- [ ] Nice to use `after_save` but it doesn't have the `update_counter` method in the controller. It should `update_posts_counter`. Kindly fix it.

Reply:
Thank you so much